core/types/header_ext.go

```diff
@@ -67,6 +67,25 @@ func (h *HeaderExtra) UnmarshalJSON(eth *ethtypes.Header, input []byte) error {
 	return nil
 }
 
+func (h *HeaderExtra) Copy(header *Header) *Header {
+	extraCopy := &HeaderExtra{
+		ExtDataHash: h.ExtDataHash,
+	}
+
+	if h.BlockGasCost != nil {
+		extraCopy.BlockGasCost = big.NewInt(0)
+		extraCopy.BlockGasCost.SetBytes(h.BlockGasCost.Bytes())
+	}
+
+	if h.ExtDataGasUsed != nil {
+		extraCopy.ExtDataGasUsed = big.NewInt(0)
+		extraCopy.ExtDataGasUsed.SetBytes(h.ExtDataGasUsed.Bytes())
+	}
+
+	cpy := ethtypes.CopyEthHeader(header)
```

Comment:
I chose to export the original eth `CopyHeader` as `CopyEthHeader`, otherwise we would have to re-implement here (I tried to unexport it as `copyHeader` before this change).